10.9-libcxx/stable/main/finkinfo/libs/pythonmods/pytest-py.info

```diff
@@ -43,11 +50,17 @@ PatchScript: <<
 
 CompileScript: %p/bin/python%type_raw[python] setup.py build
 
+# tox will download and pip-install a gazillion of extra deps, even those already installed as packages.
```

Comment:
This is for future reference: tox will try to run the generic executable `pytest` when running these tests, so it'll grab whatever version is installed via update-alternatives. This patch is needed to make it try to use the already installed pytest command for the right variant:
````
perl -pi -e 's|pytest --lsof|%p/bin/pytest-%type_raw[python] --lsof|g' tox.ini
perl -pi -e 's|pytest -n3|%p/bin/pytest-%type_raw[python] -n3|g' tox.ini
perl -pi -e 's|pytest -n8|%p/bin/pytest-%type_raw[python] -n8|g' tox.ini
perl -pi -e 's|pytest -ra|p%p/bin/ytest-%type_raw[python] -ra|g' tox.ini
perl -pi -e 's|pytest --doctest|%p/bin/pytest-%type_raw[python] --doctest|g' tox.ini
perl -pi -e 's|sphinx-build|%p/bin/sphinx-build%type_raw[python]|g' tox.ini
perl -pi -e 's|coverage run|%p/bin/coverage%type_raw[python] run|g' tox.ini
perl -pi -e 's|coverage report|%p/bin/coverage%type_raw[python] report|g' tox.ini
````
This avoids downloading some of the pip-install dependencies, but it needs pytest itself to already be installed (wonder if you can play with symlinks, PATH, and PYTHONPATH during TestScript to make an appropriate `pytest` in %b...)
!lart bad test setup.

Reply:
As far as I can see too is already doing this - using the `pytest` executable in `%b/.tox/py%type_pkg[python]/bin`, which I believe is installed from its own distribution previously created in `%b/.tox/dist/pytest-3.6.2.zip`.
Compared to this default
``` 
py36 installdeps: hypothesis>=3.56, nose, mock, requests
py36 inst: /scratch.noindex/fink.build/pytest-py36-3.6.2-1/pytest-3.6.2/.tox/dist/pytest-3.6.2.zip
py36 installed: atomicwrites==1.1.5,attrs==18.1.0,certifi==2018.4.16,chardet==3.0.4,coverage==4.5.1,hypothesis==3.66.14,idna==2.7,mock==2.0.0,more-itertools==4.2.0,nose==1.3.7,pbr==4.2.0,pluggy==0.6.0,py==1.5.4,pytest==3.6.2,requests==2.19.1,six==1.11.0,urllib3==1.23
py36 runtests: PYTHONHASHSEED='3592075579'
py36 runtests: commands[0] | pytest --lsof -ra testing
===================================================================== test session starts ======================================================================
platform darwin -- Python 3.6.6, pytest-3.6.2, py-1.5.4, pluggy-0.6.0
rootdir: /scratch.noindex/fink.build/pytest-py36-3.6.2-1/pytest-3.6.2, inifile: tox.ini
plugins: hypothesis-3.66.14
collected 2115 items

testing/acceptance_test.py .................................................x................                                                            [  3%]
testing/deprecated_test.py ..............                                                                                                                [  3%]
...
============================================== 2024 passed, 80 skipped, 11 xfailed, 3 warnings in 243.87 seconds ===============================================

```
which can n.b. not be using `/sw/etc/alternatives/pytest@ -> /sw/bin/pytest-3.7`, since there is no `hypothesis-py37` installed,
forcing `tox.ini` to use the already installed `/sw/bin/pytest-py%type-raw[python]` lets it load a number of additional plugins, collection more tests but with many more failing:
```
py36 installdeps: hypothesis>=3.56, nose, mock, requests
py36 inst: /scratch.noindex/fink.build/pytest-py36-3.6.2-1/pytest-3.6.2/.tox/dist/pytest-3.6.2.zip
py36 installed: atomicwrites==1.1.5,attrs==18.1.0,certifi==2018.4.16,chardet==3.0.4,coverage==4.5.1,hypothesis==3.66.14,idna==2.7,mock==2.0.0,more-itertools==4.2.0,nose==1.3.7,pbr==4.2.0,pluggy==0.6.0,py==1.5.4,pytest==3.6.2,requests==2.19.1,six==1.11.0,urllib3==1.23
py36 runtests: PYTHONHASHSEED='2390453554'
py36 runtests: commands[0] | /sw/bin/pytest-3.6 --lsof -ra testing
===================================================================== test session starts ======================================================================
platform darwin -- Python 3.6.6, pytest-3.6.2, py-1.5.4, pluggy-0.6.0
PyQt5 5.10.1 -- Qt runtime 5.7.1 -- Qt compiled 5.7.1
Matplotlib: 2.2.2
Freetype: 2.6.1
rootdir: /scratch.noindex/fink.build/pytest-py36-3.6.2-1/pytest-3.6.2, inifile: tox.ini
plugins: remotedata-0.2.0, relaxed-1.1.4, qt-2.3.1, openfiles-0.2.0, mpl-0.9, mock-1.10.0, doctestplus-0.1.2, arraydiff-0.2, hypothesis-3.65.1
collected 2654 items

testing/acceptance_test.py ......F.......................................FF.x............F.........F......................F................FF.x........F [  2%]
...F...                                                                                                                                                  [  2%]
testing/deprecated_test.py ............................
```

I don't see the patch to `tox.ini` reducing the number of downloaded packages, but it introduces loads of test failures (and is using whatever pytest version is already installed, not what has just been built).